File: civitasJAVA/src/civitas/CasillaJuez.java
package civitas;

import java.util.ArrayList;

/**
 * @author Jesús Navarro Merino
 * @author José Alberto Gómez García
 */

public class CasillaJuez extends Casilla{

    private static int carcel;
    
    public CasillaJuez(String nombre_introducido,int num_casilla_carcel){
        super(nombre_introducido);
        carcel=num_casilla_carcel;
    }
    
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador>todos){
        if(jugadorCorrecto(actual, todos)){
            super.informe(actual, todos);
            todos.get(actual).encarcelar(carcel);
        }
    }
    
    //El super indica que usamos un metodo de la clase padre; pero realmente actua como this: es el propio objeto, 
    //por lo que al usar el toString() de la clase padre, donde pone "this.getClass().getName()" saldrá el nombre de esta clase (CasillaJuez)
    @Override
    public String toString(){ 
        return super.toString()     
               + "\n\t Carcel: " + carcel;
    }

} //Fin de la clase CasillaJuez

File: civitasRUBY/lib/casilla_juez.rb
# encoding:utf-8

module Civitas
  class CasillaJuez < Casilla
    
    attr_reader :carcel, :nombre
    
    def initialize(_nombre,_carcel)
      super(_nombre)
      @carcel=_carcel
    end
    
    def recibeJugador(actual,todos)
        if(jugadorCorrecto(actual,todos))
            informe(actual,todos)
            todos[actual].encarcelar(@carcel)
        end
    end
    
    def toString
      super
    end
    
    def self.constructorCasillaJuez(_nombre,_carcel)
      new(_nombre,_carcel)
    end

  end
end
